doc/src/build/package-upgrades.md

```diff
@@ -0,0 +1,283 @@
+
+## Upgrading
+
+Use the `sui client upgrade` command to upgrade packages that meet the previous requirements, providing values for the following flags:
+
+* `--gas-budget`: The maximum number of gas units that can be expended before the network cancels the transaction.
+* `--cap`: The ID of the `UpgradeCap`. You receive this ID as a return from the publish command.
+
+Developers upgrading packages using Move code have access to types and functions to define custom upgrade policies. For example, a Move developer might want to disallow upgrading a package, regardless of the current package owner. The [`make_immutable` function](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/docs/package.md#0x2_package_make_immutable) is available to them to create this behavior. More advanced policies using available types like `UpgradeTicket` and `Upgrade Receipt` are also possible. For an example, see this [custom upgrade policy](https://github.com/MystenLabs/sui/issues/2045#:~:text=Implement%20a%20custom%20upgrade%20policy) on GitHub.
+
+When you use the Sui Client CLI, the `upgrade` command handles generating the upgrade digest, authorizing the upgrade with the `UpgradeCap` to get an `UpgradeTicket`, and updating the `UpgradeCap` with the `UpgradeReceipt` after a successful upgrade. To learn more about these processes, see the Move documentation for the [package module](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/docs/package.md). 
+
+## Example
+
+You develop a package named `sui_package`. Its manifest looks like the following:
+
+```move
+[package]
+name = "sui_package"
+version = "0.0.0"
+
+[addresses]
+sui_package = "0x0"
+```
+
+When your package is ready, you publish it:
+
+```shell
+sui client publish --gas-budget <GAS-BUDGET-AMOUNT>
+```
+And receive the response:
+
+```shell
+INCLUDING DEPENDENCY Sui
+INCLUDING DEPENDENCY MoveStdlib
+BUILDING MyFirstPackage
+Successfully verified dependencies on-chain against source.
+----- Transaction Digest ----
+2bn3EtHvbVY4bM1887MvFiGWnqq1YZ2RKmbeK7TrRbLL
+----- Transaction Data ----
+Transaction Signature: [Signature(Ed25519SuiSignature(Ed25519SuiSignature([0, 156, 133, 71, 156, 44, 204, 30, 31, 250, 204, 247, 60, 212, 249, 61, 112, 249, 148, 180, 83, 207, 236, 58, 99, 134, 5, 174, 115, 226, 41, 139, 192, 1, 183, 133, 38, 73, 254, 205, 190, 54, 210, 112, 144, 204, 137, 3, 8, 30, 165, 147, 120, 199, 227, 119, 53, 208, 28, 101, 34, 239, 102, 210, 1, 103, 111, 108, 165, 156, 100, 95, 13, 236, 27, 13, 127, 150, 50, 47, 155, 217, 27, 164, 61, 245, 254, 81, 182, 121, 231, 58, 150, 214, 46, 27, 222])))]
+Transaction Kind : Programmable
+Inputs: [Pure(SuiPureValue { value_type: Some(Address), value: "<PUBLISHER-ID>" })]
+Commands: [
+  Publish(_,,0x00000000000000000000000000000000000000000000000000000000000000010x0000000000000000000000000000000000000000000000000000000000000002),
+  TransferObjects([Result(0)],Input(0)),
+]
+
+Sender: <PUBLISHER-ID>
```

Comment:
isn't this the address of the package owner? Is it confusing to introduce a new ID type in this topic that is used only in this topic?